src/components/Agents/components/NewsletterModal/hooks/useDisableScroll.ts

```diff
@@ -0,0 +1,19 @@
+import { useEffect } from 'react';
+
+type UseDisableScrollArgs = {
+  condition: boolean;
+};
+
+export const useDisableScroll = ({ condition }: UseDisableScrollArgs) => {
+  useEffect(() => {
+    if (!condition) {
+      return;
+    }
+
+    document.body.style.overflowY = 'hidden';
+
+    return () => {
+      document.body.style.overflowY = 'auto';
```

Comment:
Let's reset it to its original state by removing the style property altogether. It'll help prevent it from overriding some default styles, or conflicting with other concurrent processes etc.

```suggestion
      document.body.style.overflowY = '';
```